lib/matplotlib/figure.py

```diff
@@ -125,12 +125,10 @@ def add(self, key, a):
         except TypeError:
             raise ValueError("first argument, %s, is not a valid key" % key)
 
-        a_existing = self.get(key)
```

Comment:
refer to this https://github.com/JunTan/matplotlib/commit/db343c3574c1dabd2854ead7b3272c380a43243f#diff-c3ce78ddcc28c54e04ade080c149c1b1R130

There was a `stack.remove()` attempts to remove the key-value pair if the key is added to the stack again. `stack.remove()` takes in `(key, (self.ind, axes))` as its parameter.Before casting, `stack.remove()` takes in `(key, , axes)`. Thus the casting fixes this problem. 

Then we decide to remove `stack.remove()` because the old key-value pair should not be removed or changed if the key is added again. Now it just raises error if you try to add an old key again. 

Thus the casting fixes the bug of the call to `stack.remove()`.